src/controllers/auth.js

```diff
@@ -67,7 +84,64 @@ export const register = async (req, res) => {
       res,
       StatusCodes.INTERNAL_SERVER_ERROR,
       null,
-      error.message
+      error.message,
     );
   }
 };
+
+export const ApproveCompanyActivateUser = async (req, res) => {
+  const { id } = req.params;
+  const { status } = req.body;
+  const user = await findUserById(id);
+  if (!user)
+    return formatResponse(res, StatusCodes.NOT_FOUND, null, "User not found");
+  const company = await getCompanyWhere({ userId: user.dataValues.id });
+  if (!company)
+    return formatResponse(
+      res,
+      StatusCodes.NOT_FOUND,
+      null,
+      "Company not found",
+    );
+  if (status === "approved") {
+    // If status is "approved," update user to be active and company to be active.
+    const updatedUser = await updateUser(id, {
+      isActive: true,
+    });
+    const updatedCompany = await updateCompany(company.id, {
+      status: "active",
+    });
+    if (!updatedUser || !updatedCompany)
+      return formatResponse(
+        res,
+        StatusCodes.BAD_REQUEST,
+        null,
+        "Unable to update user and company",
+      );
+  } else if (status === "deny") {
+    const updatedUser = await updateUser(id, {
+      isActive: false,
+    });
+    const updatedCompany = await updateCompany(company.id, {
+      status: "inactive",
+    });
+    if (!updatedUser || !updatedCompany)
+      return formatResponse(
+        res,
+        StatusCodes.BAD_REQUEST,
+        null,
+        "Unable to update user and company",
+      );
```

Comment:
We have too many duplicates here. Can we make it better? What do you think?

Reply:
resolved